runtime/compiler/aarch64/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -36,8 +40,1040 @@ void J9::ARM64::AheadOfTimeCompile::processRelocations()
+         }
+         break;
+
+      case TR_ValidateClassInfoIsInitialized:
+         {
+         TR::ClassInfoIsInitialized *record = reinterpret_cast<TR::ClassInfoIsInitialized *>(relocation->getTargetAddress());
+
+         cursor -= sizeof(TR_RelocationRecordBinaryTemplate);
+
+         TR_RelocationRecordValidateClassInfoIsInitializedBinaryTemplate *binaryTemplate =
+               reinterpret_cast<TR_RelocationRecordValidateClassInfoIsInitializedBinaryTemplate *>(cursor);
+
+         binaryTemplate->_classID = symValManager->getIDFromSymbol(static_cast<void *>(record->_class));
+         binaryTemplate->_isInitialized = record->_isInitialized;
+
+         cursor += sizeof(TR_RelocationRecordValidateClassInfoIsInitializedBinaryTemplate);
+         }
+         break;
+
+      case TR_SymbolFromManager:
+         {
+         uint8_t *symbol = relocation->getTargetAddress();
+         uint16_t symbolID = comp->getSymbolValidationManager()->getIDFromSymbol(static_cast<void *>(symbol));
+
+         uint16_t symbolType = (uint16_t)(uintptrj_t)relocation->getTargetAddress2();
+
+         cursor -= sizeof(TR_RelocationRecordBinaryTemplate);
+
+         TR_RelocationRecordSymbolFromManagerBinaryTemplate *binaryTemplate =
+               reinterpret_cast<TR_RelocationRecordSymbolFromManagerBinaryTemplate *>(cursor);
+
+         binaryTemplate->_symbolID = symbolID;
+         binaryTemplate->_symbolType = symbolType;
+
+         cursor += sizeof(TR_RelocationRecordSymbolFromManagerBinaryTemplate);
+         }
+         break;
+
+      case TR_DiscontiguousSymbolFromManager:
+         {
+         uint8_t *symbol = (uint8_t *)relocation->getTargetAddress();
+         uint16_t symbolID = comp->getSymbolValidationManager()->getIDFromSymbol(static_cast<void *>(symbol));
+
+         uint16_t symbolType = (uint16_t)(uintptrj_t)relocation->getTargetAddress2();
+
+         cursor -= sizeof(TR_RelocationRecordBinaryTemplate);
+
+         TR_RelocationRecordSymbolFromManagerBinaryTemplate *binaryTemplate =
+               reinterpret_cast<TR_RelocationRecordSymbolFromManagerBinaryTemplate *>(cursor);
+
+         binaryTemplate->_flags |= (flags & RELOCATION_RELOC_FLAGS_MASK);
+         binaryTemplate->_symbolID = symbolID;
+         binaryTemplate->_symbolType = symbolType;
+
+         cursor += sizeof(TR_RelocationRecordSymbolFromManagerBinaryTemplate);
+         }
+         break;
+
+      case TR_ValidateClass:
+         {
+         *(uintptrj_t*)cursor = (uintptrj_t)relocation->getTargetAddress(); // Inlined site index
+         cursor += SIZEPOINTER;
+
+         TR::AOTClassInfo *aotCI = (TR::AOTClassInfo*)relocation->getTargetAddress2();
+         *(uintptrj_t*)cursor = (uintptrj_t) aotCI->_constantPool;
+         cursor += SIZEPOINTER;
+
+         *(uintptrj_t*)cursor = (uintptrj_t) aotCI->_cpIndex;
+         cursor += SIZEPOINTER;
+
+         uintptr_t classChainOffsetInSharedCache = sharedCache->offsetInSharedCacheFromPointer(aotCI->_classChain);
```

Comment:
If https://github.com/eclipse/openj9/pull/7379 gets merged first, this will need to change to `self()->offsetInSharedCacheFromPointer(sharedCache, aotCI->_classChain);`

Reply:
Assuming #7379 gets merged first, I changed all the calls to `offsetInSharedCacheFromPointer` as proposed.